vmdk_plugin/drivers/vmdk/vmdk_driver.go

```diff
@@ -400,7 +400,11 @@ func (d *VolumeDriver) Mount(r volume.MountRequest) volume.Response {
 			log.Fields{"name": r.Name, "error": err.Error()},
 		).Error("Failed to mount ")
 
-		d.decrRefCount(r.Name)
+		refcnt, _ := d.decrRefCount(r.Name)
+		if refcnt == 0 {
+			log.Infof("Detaching %s - it is not used anymore", r.Name)
+			d.ops.Detach(r.Name, nil) // try to detach before failing the request for volume
```

Comment:
Logically we do not want to unmount since we know the mount has failed. We just want to clean up the attach.
Technically UnmountVolume does a set of useless calls (i.e. tries to locate the name and unmount volume) which we know will fail  and  guarantee confusing messages about failing to unmount, and  then it will do the only useful step (d.ops.Detach) which we can do directly.  

I guess that was a long winded version of "no. we need to call Detach"  :-)